src/ParticleSpecies.cpp

```diff
@@ -111,12 +111,11 @@ void ParticleSpecies::read()
             rc.written() = true;
             r.read();
         }
-        catch (std::runtime_error const &)
+        catch (error::ReadError const &err)
         {
-            std::cerr << "WARNING: Skipping invalid openPMD record '"
-                      << record_name << "'" << std::endl;
-            while (!IOHandler()->m_work.empty())
-                IOHandler()->m_work.pop();
```

Comment:
No need to clear the queue here anymore, it's already cleared by `AbstractIOHandler::flush()` upon failure